src/main/java/ch/njol/skript/expressions/ExprAffectedEntities.java

```diff
@@ -58,20 +61,52 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 		return true;
 	}
 
-	@Nullable
 	@Override
-	protected LivingEntity[] get(Event e) {
-		if (e instanceof AreaEffectCloudApplyEvent)
-			return ((AreaEffectCloudApplyEvent) e).getAffectedEntities().toArray(new LivingEntity[0]);
+	protected LivingEntity @Nullable [] get(Event event) {
+		if (event instanceof AreaEffectCloudApplyEvent areaEvent)
+			return areaEvent.getAffectedEntities().toArray(new LivingEntity[0]);
 		return null;
 	}
 
-	@Nullable
 	@Override
-	public Iterator<? extends LivingEntity> iterator(Event e) {
-		if (e instanceof AreaEffectCloudApplyEvent)
-			return ((AreaEffectCloudApplyEvent) e).getAffectedEntities().iterator();
-		return super.iterator(e);
+	public @Nullable Iterator<? extends LivingEntity> iterator(Event event) {
+		if (event instanceof AreaEffectCloudApplyEvent areaEvent)
+			return areaEvent.getAffectedEntities().iterator();
+		return super.iterator(event);
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+        if (mode == ChangeMode.REMOVE_ALL)
+            return null;
+		return CollectionUtils.array(LivingEntity[].class);
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		if (!(event instanceof AreaEffectCloudApplyEvent areaEvent))
+			return;
+
+		LivingEntity[] entities = (LivingEntity[]) delta[0];
```

Comment:
```suggestion
		LivingEntity[] entities = (LivingEntity[]) delta;
```
delta as a whole is the values to use to change, there won't be nested arrays.